### PR TITLE
Add functionality to create batch in wholesale

### PR DESCRIPTION
**Pull request by BjarkeMeier**
## References

- Issue [here](https://github.com/Energinet-DataHub/opengeh-wholesale/issues/33)

### CODE REVIEW

File: libs/dh/wholesale/data-access-api/src/lib/dh-wholesale-data-access-api.store.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Injectable } from '@angular/core';
+import { ComponentStore, tapResponse } from '@ngrx/component-store';
+import { filter, map, Observable, switchMap, tap } from 'rxjs';
```
Comment (dzhavat) [inline libs/dh/wholesale/data-access-api/src/lib/dh-wholesale-data-access-api.store.ts:19]: suggestion: remove this line. All of the imports are not used anywhere in the file.

File: libs/gf/configuration-danish-locale/project.json [hunk 2]
```diff
@@ -1,6 +1,5 @@
 {
   "projectType": "library",
-  "root": "libs/gf/configuration-danish-locale",
```
Comment (dzhavat) [inline libs/gf/configuration-danish-locale/project.json:3]: question: do you remember changing this line? It doesn't look like it's something that should be part of this PR. Can be reverted.

File: libs/dh/wholesale/data-access-api/src/lib/dh-wholesale-data-access-api.store.ts (hunk 1 above)
Comment (dzhavat) [reply libs/dh/wholesale/data-access-api/src/lib/dh-wholesale-data-access-api.store.ts:19]: Actually I've made a mistake. Only `filter`, `map` and `tap` are not used. They can be removed

File: libs/gf/configuration-danish-locale/project.json (hunk 2 above)
Comment (BjarkeMeier) [reply libs/gf/configuration-danish-locale/project.json:3]: Thx. Removed.

File: libs/dh/wholesale/data-access-api/src/lib/dh-wholesale-data-access-api.store.ts (hunk 1 above)
Comment (BjarkeMeier) [reply libs/dh/wholesale/data-access-api/src/lib/dh-wholesale-data-access-api.store.ts]: Thx. Removed.

File: apps/dh/api-dh/source/DataHub.WebApi/Controllers/Wholesale/WholesaleBatchRequestDto.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Collections.Generic;
+
+namespace Energinet.DataHub.WebApi.Controllers.Wholesale
+{
+    public sealed record WholesaleBatchRequestDto(WholesaleProcessType ProcessType, IEnumerable<string> GridAreaCodes);
```
Comment (lasrinnil) [inline apps/dh/api-dh/source/DataHub.WebApi/Controllers/Wholesale/WholesaleBatchRequestDto.cs:19]: newline each parameter for easier reading

File: apps/dh/api-dh/source/DataHub.WebApi/HttpClientFactory.cs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+        private void SetAuthorizationHeader(HttpClient httpClient)
+        {
+            var authHeaderValue = GetAuthorizationHeaderValue();
+            httpClient.DefaultRequestHeaders.Add("Authorization", authHeaderValue);
+        }
+
+        private string GetAuthorizationHeaderValue()
+        {
+            return _httpContextAccessor.HttpContext!.Request.Headers
```
Comment (lasrinnil) [inline apps/dh/api-dh/source/DataHub.WebApi/HttpClientFactory.cs:52]: Could we get rid of the null forgiver?

File: build/ci/dh/api-dh/ci.appsettings.Development.json [hunk 5]
```diff
@@ -8,7 +8,8 @@
   },
   "ApiClientSettings": {
     "MeteringPointBaseUrl": "https://app-webapi-mpt-u-001.azurewebsites.net",
-    "ChargesBaseUrl": "https://app-webapi-charges-u-001.azurewebsites.net"
+    "ChargesBaseUrl": "https://app-webapi-charges-u-001.azurewebsites.net",
+    "WholesaleBaseUrl": "https://app-webapi-wholsal-u-001.azurewebsites.net"
   },
   "B2C_TENANT_ID" : "4a7411ea-ac71-4b63-9647-b8bd4c5a20e0",
```
Comment (lasrinnil) [inline build/ci/dh/api-dh/ci.appsettings.Development.json:14]: Im not sure but is it ok exposing these values?

Comment (djorgensendk) [reply build/ci/dh/api-dh/ci.appsettings.Development.json:14]: @lasrinnil I agree, and the values can be fetched from key vault secrets

Comment (BjarkeMeier) [reply build/ci/dh/api-dh/ci.appsettings.Development.json:14]: @djorgensendk could you manage this in MD? It's based on what has already been done and seems to be based on thoughts by Texas Rangers (maintaining and copying this file is described in the repo gettting started documentation).

File: apps/dh/api-dh/source/DataHub.WebApi/HttpClientFactory.cs (hunk 4 above)
Comment (BjarkeMeier) [reply apps/dh/api-dh/source/DataHub.WebApi/HttpClientFactory.cs:52]: I don't think so. How would you do that?
We could wrap the usage of the HttpContextAccessor but that would just move the problem.

File: apps/dh/api-dh/source/DataHub.WebApi/Controllers/Wholesale/WholesaleProcessType.cs [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Energinet.DataHub.WebApi.Controllers.Wholesale
+{
+    public enum WholesaleProcessType
+    {
+        BalanceFixing,
```
[the remark this answers is not in the document]
Comment (BjarkeMeier) [reply apps/dh/api-dh/source/DataHub.WebApi/Controllers/Wholesale/WholesaleProcessType.cs:19]: For now, we're using the name. But let's remember your comment on the next iteration of this code.

**Pull request merged**